cargo-dist/src/linkage.rs

```diff
@@ -393,8 +393,13 @@ fn do_otool(path: &Utf8PathBuf) -> DistResult<Vec<String>> {
 fn do_ldd(path: &Utf8PathBuf) -> DistResult<Vec<String>> {
     let mut libraries = vec![];
 
+    // We ignore the status here because for whatever reason arm64 glibc ldd can decide
+    // to return non-zero status on binaries with no dynamic linkage (e.g. musl-static).
+    // This was observed both in arm64 ubuntu and asahi (both glibc ldd).
+    // x64 glibc ldd is perfectly fine with this and returns 0, so... *shrug* compilers!
     let output = Cmd::new("ldd", "get linkage info from ldd")
         .arg(path)
+        .check(false)
```

Comment:
i will ask just for completions sake- can this command return an error we would ever want to be a hard fail? thinking it through it briefly on my own, i think no- but feel like we should have the conversation just to be sure.

Reply:
so i'm actually restoring old behaviour (we were ignoring these statuses until i landed axoprocess, because i noticed we were and misty thought it wasn't intentional). we could use non-zero as the same thing we do for "dynamically linked", but at this point i don't trust this exit code, so i'm inclined for us to just always try to parse the output.